provider-service/base/pkg/streams/stream.go

```diff
@@ -0,0 +1,30 @@
+package streams
+
+type Inlet[I any] interface {
+	In() chan<- I
+}
+
+type Outlet[O any] interface {
+	Out() <-chan O
+}
+
+type ErrorOutlet[E any] interface {
```

Comment:
wouldn't an error outlet/inlet always be of type `error`

Reply:
not necessarily, it could be the original message coupled with the error as an example

Reply:
but it would always implement the `error` interface